### PR TITLE
New sign up confirmation logic

### PR DESCRIPTION
**Pull request by Greengrer**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/web/controller/RestSignupController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        User byEmail = userService.findByEmail(username);
+        return Map.of(
+                "user", userMapper.mapUserToDto(byEmail),
+                "token", token);
+    }
+
+    private Map<String, String> getStatusInfoMap(String code, String s) {
+        return Map.of(
+                "resultCode", code,
+                "message", s);
```
Comment (TarasOrlovskyi) [inline src/main/java/com/vinylteam/vinyl/web/controller/RestSignupController.java]: please, rename 's'

File: src/main/java/com/vinylteam/vinyl/exception/entity/EmailConfirmationErrors.java [hunk 2]
```diff
@@ -7,8 +7,8 @@
 @Getter
 public enum EmailConfirmationErrors {
 
-    TOKEN_FROM_LINK_NOT_FOUND("Sorry, we can't find this link. Would you like to request another?"),
-    TOKEN_FROM_LINK_NOT_UUID("Sorry, this is not a valid link"),
+    TOKEN_FROM_LINK_NOT_FOUND("your link isn't correct. Please, try again or contact us"),
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/exception/entity/EmailConfirmationErrors.java]: your not Your?

File: src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    private ZoneOffset offset = ZoneOffset.ofTotalSeconds(0);
+
+                    @Override
+                    public long expireAfterCreate(@NonNull String s, @NonNull LocalDateTime localDateTime, long l) {
+                        return getDifference(localDateTime);
+                    }
+
+                    @Override
+                    public long expireAfterUpdate(@NonNull String s, @NonNull LocalDateTime localDateTime, long l, @NonNegative long l1) {
+                        return l1;
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java]: magic number

File: src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                    }
+
+                    @Override
+                    public long expireAfterRead(@NonNull String s, @NonNull LocalDateTime localDateTime, long l, @NonNegative long l1) {
+                        return l1;
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java]: magic number

File: src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                    }
+
+                    private long getDifference(@NonNull LocalDateTime localDateTime) {
+                        return 1_000_000_000L * (localDateTime.toEpochSecond(offset) - LocalDateTime.now().toEpochSecond(offset));
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java]: magic number

File: src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Autowired
+    public void setJwtService(JwtService jwtService) {
+        this.jwtService = jwtService;
+    }
+
+    @PostConstruct
+    public void initCache() {
+        logoutTokensCache = Caffeine.newBuilder()
+                .expireAfter(new Expiry<String, LocalDateTime>() {
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java]: Encapsulate these into private method and use method's reference instead

File: src/main/java/com/vinylteam/vinyl/service/JwtTokenProvider.java [hunk 7]
```diff
@@ -47,22 +58,27 @@ public boolean validateToken(String token) {
...
-                    .setSigningKey(secretKey)
-                    .build()
-                    .parseClaimsJws(token);
+            Jws<Claims> claims = getClaims(token);
 
             if (claims.getBody().getExpiration().before(new Date())) {
                 return false;
             }
-            return true;
+            return true && !tokenStorageService.isTokenBlocked(token);
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/service/JwtTokenProvider.java]: really?

File: src/main/java/com/vinylteam/vinyl/service/JwtTokenProvider.java [hunk 8]
```diff
@@ -93,6 +109,16 @@ public Authentication getAuthentication(String token) {
         return new UsernamePasswordAuthenticationToken(userDetails, "", userDetails.getAuthorities());
     }
 
+    @Override
+    public LocalDateTime getExpirationDate(String token) {
+        Jws<Claims> claims = getClaims(token);
+
+        Date expirationDate = claims.getBody().getExpiration();
+        return expirationDate.toInstant()
+                .atZone(ZoneId.systemDefault())
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/service/JwtTokenProvider.java]: why system default not exact zone?

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultEmailConfirmationService.java [hunk 9]
```diff
@@ -52,12 +52,20 @@
...
+                .token(generateToken())
                 .build();
+        log.info("Generated token: {}", newConfirmationToken.getToken());
         confirmationTokenDao.add(newConfirmationToken);
         return newConfirmationToken;
     }
 
+    @Override
+    public UUID generateToken() {
+
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultEmailConfirmationService.java]: obsolete extra line

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java [hunk 10]
```diff
@@ -133,9 +137,9 @@ public void signInCheck(UserInfoRequest userProfileInfo) {
         if (!securityService.validateIfPasswordMatches(userToCheckAgainst, userProfileInfo.getPassword().toCharArray())) {
             throw new RuntimeException(UserErrors.WRONG_CREDENTIALS_ERROR.getMessage());
         }
-        if (!userToCheckAgainst.getStatus()) {
+/*        if (!userToCheckAgainst.getStatus()) {
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java]: YAGNI

File: src/main/java/com/vinylteam/vinyl/web/controller/EmailConfirmationController.java [hunk 11]
```diff
@@ -32,4 +39,21 @@ public ModelAndView getConfirmationPage(@RequestParam(value = "token") String to
         return modelAndView;
     }
 
+
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/web/controller/EmailConfirmationController.java]: Obsolete extra lines

File: src/main/java/com/vinylteam/vinyl/web/controller/SignInController.java [hunk 12]
```diff
@@ -9,15 +9,14 @@
...
-import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.servlet.ModelAndView;
 
 import javax.servlet.http.HttpSession;
 
 @Slf4j
-@Controller
+//@Controller
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/web/controller/SignInController.java]: YAGNI

File: src/main/resources/application.properties [hunk 13]
```diff
@@ -1,15 +1,21 @@
...
+#spring.datasource.driver-class-name=org.postgresql.Driver
 spring.datasource.hikari.maximum.pool.size=5
 #Path to resource folder
 resource.path=static
 #Logging properties
 PACKAGE_NAME=logs
 FILE_NAME=vinyl.log
 FILE_NAME_PATTERN=vinyl-%d{yyyy-MM-dd}.%i.log
-FILTER_LEVEL=info
+FILTER_LEVEL=debug
```
Comment (hemiSRT8) [inline src/main/resources/application.properties]: sure?

**Review by hemiSRT8 — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/web/handler/ApiExceptionHandler.java [hunk 14]
```diff
@@ -5,15 +5,17 @@
 import org.springframework.web.bind.annotation.ExceptionHandler;
 
 import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
 
 //@ControllerAdvice
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/web/handler/ApiExceptionHandler.java]: YAGNI

File: src/test/java/com/vinylteam/vinyl/web/controller/SignupControllerITest.java [hunk 15]
```diff
@@ -14,32 +18,40 @@
...
 import static org.mockito.Mockito.when;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
 
 @SpringBootTest
 @AutoConfigureMockMvc
 @Slf4j
-class SignupControllerITest {
+class RestSignupControllerITest {
```
Comment (hemiSRT8) [inline src/test/java/com/vinylteam/vinyl/web/controller/SignupControllerITest.java]: without REST

**Review by hemiSRT8 — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java (hunk 4 above)
Comment (littlepony99) [reply src/main/java/com/vinylteam/vinyl/security/InMemoryLogoutTokenStorageService.java]: code from previous PR

File: src/main/java/com/vinylteam/vinyl/web/controller/EmailConfirmationController.java (hunk 11 above)
Comment (littlepony99) [reply src/main/java/com/vinylteam/vinyl/web/controller/EmailConfirmationController.java]: code from previous PR

**Review by littlepony99 — APPROVED** [review]

**Review by hemiSRT8 — APPROVED** [review]

**Pull request merged**